Commit teste

dani-barnasque opened the pull request.

dani-barnasque committed:
Commit teste

doniniramos commented on lib/main.dart:
```diff
@@ -0,0 +1,182 @@
+import 'dart:convert';
+import 'dart:core';
+import 'dart:async';
+import 'package:flutter/cupertino.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter/widgets.dart';
+import 'package:http/http.dart' as http;
+
```
**[NIT](https://stackoverflow.com/questions/27810522/what-does-nit-mean-in-hacker-speak)**: Para facilitar a leitura e organização separamos os imports em chunks de `dart`, `flutter`, dependências externas ex: `http` e arquivos do projeto `lib/...` e dentro de cada chunk **TODOS** em ordem alfabética. No teu caso o apropriado seria assim:

```
import 'dart:async';
import 'dart:convert';
import 'dart:core';

import 'package:flutter/cupertino.dart';
import 'package:flutter/material.dart';
import 'package:flutter/widgets.dart';

import 'package:http/http.dart' as http;

dani-barnasque replied:
Já sabia que era em ordem alfabética, mas não sabia dos chunks. Valeu!

dani-barnasque replied:
essa config que tu me sugeriu é no androidstudio né? Pq eu uso VS Code e n achei

doniniramos replied:
Tenta [esse link](https://stackoverflow.com/questions/39494277/how-do-you-format-code-on-save-in-vs-code)

doniniramos commented on lib/main.dart:
```diff
@@ -0,0 +1,182 @@
+          primaryColor: Colors.white,
+          inputDecorationTheme: InputDecorationTheme(
+            enabledBorder:
+                OutlineInputBorder(borderSide: BorderSide(color: Colors.white)),
```
Nit: vírgula
`OutlineInputBorder(borderSide: BorderSide(color: Colors.white),),`

dani-barnasque replied:
o que é Nit? hehe
beleza, sempre deixar virgula depois do último atributo então

doniniramos replied:
Com essa vírgula e aquela `config` que eu te passei, o editor `auto-indent`aria o código 😄

doniniramos commented on lib/main.dart:
```diff
@@ -0,0 +1,182 @@
+            enabledBorder:
+                OutlineInputBorder(borderSide: BorderSide(color: Colors.white)),
+            focusedBorder:
+                OutlineInputBorder(borderSide: BorderSide(color: Colors.amber)),
```
Nit: vírgula
`OutlineInputBorder(borderSide: BorderSide(color: Colors.amber),),`

doniniramos commented on lib/main.dart:
```diff
@@ -0,0 +1,182 @@
+            focusedBorder:
+                OutlineInputBorder(borderSide: BorderSide(color: Colors.amber)),
+            hintStyle: TextStyle(color: Colors.amber),
+          )),
```
Nit: vírgula
`),),`

doniniramos commented on lib/main.dart:
```diff
@@ -0,0 +1,182 @@
+    euroController.text = "";
+  }
+
+  void _realChange(String text) {
+    if (text.isEmpty) {
+      _clearAll();
+      return;
+    }
+    double real = double.parse(text);
+    dolarController.text = (real / dolar).toStringAsFixed(2);
+    euroController.text = (real / euro).toStringAsFixed(2);
+  }
+
+  void _dolarChange(String text) {
+    if (text.isEmpty) {
+      _clearAll();
+      return;
+    }
+    double dolar = double.parse(text);
+    double real = dolar * this.dolar;
+    realController.text = (real).toStringAsFixed(2);
+    euroController.text = (real / euro).toStringAsFixed(2);
+  }
+
+  void _euroChange(String text) {
+    if (text.isEmpty) {
+      _clearAll();
+      return;
+    }
+    double euro = double.parse(text);
+    double real = euro * this.euro;
+    realController.text = (real).toStringAsFixed(2);
+    dolarController.text = (real / dolar).toStringAsFixed(2);
+  }
```
Tá repetindo bastante código aí, hein, Danielzinho

Tu podia ter uma função que apenas converta uma pra outro, tipo essa, daí tu chama dela retornando o valor que tu quer
```
String _convertFromTo(double from, double to) {  
  return (from / to).toStringAsFixed(2);
}
```
A função acima é totalmente opcional, só uma sugestão pra diminuir o número de linhas que fazem a mesma coisa. 

Já o código abaixo não é legal e eu vou te dizer porque

```
    realController.text = (real).toStringAsFixed(2);
    dolarController.text = (real / dolar).toStringAsFixed(2);

  Ou

    realController.text = (real).toStringAsFixed(2);
    euroController.text = (real / euro).toStringAsFixed(2);

  Ou 

    dolarController.text = (real / dolar).toStringAsFixed(2);
    euroController.text = (real / euro).toStringAsFixed(2);
```
Porque não é legal? 
Porque tu tá alterando um valor de uma variável fora desse escopo, o que no teu caso, nem é tão grave. Mas imagina tu chamar uma função com o mesmo nome `_euroChange`, mas ela tem centenas de linhas, tu chama ela e do nada em outra parte da tela tu gera um bug, porque em outra função que tu conhece bem, a `_dolarChange`, tu chama uma função de nome `_updateInputText` que faz o update dos `Controllers.text`. E, embora semelhante, o comportamento delas é diferente, quanto tu chama a função `_updateInputText`, tu adiciona 2 casas depois da vírgula, mas esse código da função `_euroChange` tu não adiciona nenhuma. Gerando, assim, inconsistência. Porque tu está alterando o mesmo cara de lugares diferentes do teu código, de **funções que não tem necessariamente um nome claro sobre o que elas estão realmente fazendo**.

De novo, no teu caso é de boas, tem pouco código. Mas quando as coisas começam a crescer, é melhor conter esses riscos.

dani-barnasque replied:
Entendi abstratamente essa indicação sobre o risco, mas ainda não entendi na prática como isso se manifesta. Logo não consegui pensar em uma alteração para fazer aqui

doniniramos replied:
No teu caso, digamos que por acaso tu tenha deixado de inserir `.toStringAsFixed(2)` (O que não foi o caso) e na revisão eu também não percebi esse problema. O app chegaria para o usuário com bug, se tu usasse um método pra todos, tu diminui essa chance, porque daria errado em todos e alguem iria perceber antes

doniniramos commented on lib/main.dart:
```diff
@@ -0,0 +1,182 @@
+                  );
+                }
+            }
+          }),
```
Nit: vírgula
```
}),

doniniramos reviewed (changes requested):
Usa essas configurações no teu projeto, quando tu salvar ele vai indentar corretamente teu código. As vírgulas nos lugares que eu tô apontando nos `comments` abaixo também são pra melhorar a indentação

![Screen Shot 2020-07-06 at 19 07 55](https://user-images.githubusercontent.com/31694530/86676371-ea1b9380-bfd0-11ea-85c4-01eee883d127.png)

dani-barnasque committed:
- Improved organization of the imports of the project
- Improved identation
- Included Function _convertFromTo in order to improve code quality
- updated Functions _realChange, _dolarChange, _euroChange

doniniramos commented on lib/main.dart:
```diff
@@ -68,7 +73,7 @@ class _HomeState extends State<Home> {
     double dolar = double.parse(text);
     double real = dolar * this.dolar;
     realController.text = (real).toStringAsFixed(2);
```
Nit: Não precisa desses parênteses `(real)`, pode remover

doniniramos commented on lib/main.dart:
```diff
@@ -79,7 +84,11 @@ class _HomeState extends State<Home> {
     double euro = double.parse(text);
     double real = euro * this.euro;
     realController.text = (real).toStringAsFixed(2);
```
Nit: Não precisa desses parênteses `(real)`, pode remover
